### ISSUE TITLE
MacOS menu - show/hide rest of the menu from Application menu?

### ISSUE THREAD

**Issue by emendelson** [joncampbell123/dosbox-x#971]
Is there any hope of adding a Show/Hide Menu item to the macOS application menu, above the "About DOSBox-X" item? I suppose this would be added to menu_osx.mm somewhere around line 111. It would make it possible to show/hide the DOSBox-X menu without remembering the keystroke.

It isn't important, but it wouldn't hurt, and might be useful for applications where you normally don't want to show the menu, but have the cleanest possible interface.

**Comment by joncampbell123** [comment]
Yes.

The latest commit mods SDL1 so that it can re-use the Application menu it generates to re-add those commands.

SDL2 however does not give me that option.

Can you compile from source? The latest code has working touch bar support.

**Comment by emendelson** [comment]
Yes, I'm compiling from source, but I don't have a touch bar Mac. I couldn't stand the keyboard on the post-2015 models, so I'm still using early-2015 models. I had a borrowed touch bar MacBook Pro in the summer of 2017, so I must have been using it to test the beta version of High Sierra, but I haven't had it for more than a year.

**Comment by emendelson** [comment]
That is exactly what it looks like. Well done.

As always with SDL1 menus, it takes some effort to get the menu to respond, but it eventually does.

This is very good to have. What I was originally asking about was an item on the same menu that would do the same thing as:

config -set showmenu=false|true

The idea is to reduce distractions that might lead non-technical users to get into situations they don't want.

**Comment by emendelson** [comment]
Fair enough!
